merkledag/ipld.md

```diff
@@ -308,6 +308,116 @@ In order to preserve merkle-linking's power, we must ensure that there is a sing
+- When using paths as defined earlier in this document, links should be accessible without further indirection. This requires the top node object to have keys corresponding to link names.
+- Link names should not conflict with other keys.
+
+There are multiple ways to do that that will be described next.
+
+### Current encoding in go-ipld
+
+go-ipld implements the following conversion:
+
+    {
+      "<Links[0].Name.(escaped)>": {
+        "hash": "<Links[0].Hash>",
+        "name": "<Links[0].Name>",
+        "size": <Links[0].Tsize>
+      },
+      "<Links[1].Name.(escaped)>": {
+        "hash": "<Links[1].Hash>",
+        "name": "<Links[1].Name>",
+        "size": <Links[1].Tsize>
+      },
+      ...
+      "@attrs": {
+        "data": "<Data>",
+        "links": [
+          {
+            "hash": "<Links[0].Hash>",
+            "name": "<Links[0].Name>",
+            "size": <Links[0].Tsize>
+          },
+          {
+            "hash": "<Links[1].Hash>",
+            "name": "<Links[1].Name>",
+            "size": <Links[1].Tsize>
+          }
+        ]
+      }
+    }
+
+Notes :
+
+- The `links` array in the `@attrs` section is there to preserve order and duplicate links to hash back to the exact same protocol buffer object.
+
+- The link names are escaped to prevent clashing with the `@attr` key.
+
+- The escaping mechanism transforms the `@` character into `\@`. This mechanism also implies a modification of the path algorithm. When a path component contains the `@` character, it must be escaped to look it up in the IPLD Node object.
+
+    For example, a path a path `/root/first@component/second@component/third_component` would look for object `root["first\@component"]["second\@component"]["third_component"]` (following mlinks when necessary).
+
+- Links are represented using the `hash` key instead of `mlink` as used in this specification. This must be changed.
+
+### Other proposition that does away with escaping
+
+We can imagine another transformation where the link names are not escaped. For example:
+
+    {
+      "<Links[0].Name>": {
+        "mlink": "<Links[0].Hash>",
+        "tsize": <Links[0].Tsize>
+      },
+      "<Links[1].Name>": {
+        "mlink": "<Links[2].Hash>",
+        "tsize": <Links[2].Tsize>
+      },
+      ...
+      ".": {
+        "data": "<Data>",
+        "links": [
+          "<Links[0].Name>",
+          {
+            "name": "<Link[1].Name>",
+            "mlink": "<Links[1].Hash>",
+            "tsize": <Links[1].Tsize>
+          }
+          "<Links[2].Name>",
+          ...
+        ]
+      }
+    }
+
+Notes:
+
+- Very conveniently, we use the key `.` to represent data for the current node, and any other key can represent a link. This means that we forbid link to be named `.`. This is in any case a good thing to do as the `.` element in paths can always be removed (the same way `..` can be replaced by the parent directory)
```

Comment:
See PR #60 where I make a distinction between filesystem paths (for use in filesystems) and IPLD paths (used to access every corner of the object model).

This would be ok for me for the IPLD paths. But for filesystem paths, I still want to have a file named "@attrs" in a directory somewhere (or in a HTTP url) without intending a special meaning to it.